src/core/config.c

```diff
@@ -0,0 +1,137 @@
+#include "config.h"
+#include "xdpw.h"
+#include "logger.h"
+
+#include <dictionary.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include "iniparser.h"
+
+void print_config(enum LOGLEVEL loglevel, struct xdpw_config *config) {
+	logprint(loglevel, "config: Configfile %s",config->conf.configfile);
+	logprint(loglevel, "config: Loglevel %d",config->logger_conf.loglevel);
+	logprint(loglevel, "config: Outputname  %s",config->screencast_conf.output_name);
+}
+
+// NOTE: calling destroy_config won't prepare the config to be read again from config file
+// with init_config since to pointers and other values won't be reset to NULL, or 0
+void destroy_config(struct xdpw_config *config) {
+	logprint(DEBUG, "config: destroying config");
+
+	free(&config->conf.configfile);
+
+	// screencast
+	free(&config->screencast_conf.output_name);
+}
+
+void getstring_from_conffile(dictionary *d, char *key, char **dest, char *def) {
+	if (*dest != NULL) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, def);
+	if (c != NULL) {
+		// Allow keys without value as default
+		if (strcmp(c, "") != 0) {
+			*dest = strdup(c);
+		} else {
+			*dest = strdup(def);
+		}
+	}
+}
+
+void getenum_from_conffile(dictionary *d, char *key, int *dest, const char *array[] , int num_array, int def) {
+	if (*dest != 0) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, NULL);
+	if (c != NULL) {
+		for (int i = 0; i < num_array; i++) {
+			if (strcmp(c, array[i]) == 0) {
+				*dest = i;
+				return;
+			}
+		}
+	}
+	*dest = def;
+}
+
+static bool file_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+static char *config_path(char *prefix, char *filename) {
+	if (!prefix || !*prefix || !filename || !*filename) {
+		return NULL;
+	}
+
+	char *config_folder = "xdg-desktop-portal-wlr";
+
+	size_t length = 3 + strlen(prefix) + strlen(config_folder) + strlen(filename);
+	char *path = (char*)calloc(length, sizeof(char));
+	strcat(path, prefix);
```

Comment:
Can we use `snprintf` here? Too easy to trip on bufefr overflows with `strcat` .